src/akkudoktoreos/core/ems.py

```diff
@@ -265,151 +271,164 @@ def set_start_hour(self, start_hour: Optional[int] = None) -> None:
             start_datetime = to_datetime().set(hour=start_hour, minute=0, second=0, microsecond=0)
             self.set_start_datetime(start_datetime)
 
-    def simuliere_ab_jetzt(self) -> dict[str, Any]:
-        jetzt = to_datetime().now()
-        start_stunde = jetzt.hour
-        return self.simuliere(start_stunde)
+    def simulate_start_now(self) -> dict[str, Any]:
+        start_hour = to_datetime().now().hour
+        return self.simulate(start_hour)
 
-    def simuliere(self, start_stunde: int) -> dict[str, Any]:
+    def simulate(self, start_hour: int) -> dict[str, Any]:
         """hour.
 
         akku_soc_pro_stunde begin of the hour, initial hour state!
         last_wh_pro_stunde integral of  last hour (end state)
         """
         # Check for simulation integrity
         if (
-            self.gesamtlast is None
-            or self.pv_prognose_wh is None
-            or self.strompreis_euro_pro_wh is None
+            self.load_energy_array is None
```

Comment:
more than one can be missing and all should be reported:
```
    missing_data = []

    if self.load_energy_array is None:
        missing_data.append("Load Curve")
    if self.pv_prediction_wh is None:
        missing_data.append("PV Forecast")
    if self.elect_price_hourly is None:
        missing_data.append("Electricity Price")
    if self.ev_charge_hours is None:
        missing_data.append("EV Charge Hours")
    if self.ac_charge_hours is None:
        missing_data.append("AC Charge Hours")
    if self.dc_charge_hours is None:
        missing_data.append("DC Charge Hours")
    if self.elect_revenue_per_hour_arr is None:
        missing_data.append("Feed-in Tariff")

    if missing_data:
        error_msg = (
            "Mandatory data missing - "
            + ", ".join(missing_data)
        )
        logger.error(error_msg)
        raise ValueError(error_msg)
```

Reply:
this is too complicated for mypy :(

Reply:
lol ok something for later